backend-hormonia/app/middleware/csrf.py

```
@@ -232,14 +252,35 @@ async def protected_route():
+        
+        if csrf_header and not csrf_cookie:
+            # Cross-domain scenario: validate header token manually
+            logger.debug(f"CSRF validation using header-only mode for {request.url.path}")
+            
+            # Simple validation: check if token format is correct (JWT-like)
+            if len(csrf_header) > 50 and '.' in csrf_header:
+                logger.debug(f"CSRF validation successful (header-only) for {request.url.path}")
+                return  # Accept the token
+            else:
+                logger.warning(f"CSRF token format invalid for {request.url.path}")
+                raise CsrfProtectError("Invalid CSRF token format")
```

Pylint (3.3.9)

[refactor] Unnecessary "else" after "return", remove the "else" and de-indent the code inside it

(R1705)